File: Python3/1980_find_unique_binary_string.py
# https://leetcode.cn/problems/find-unique-binary-string/

class Solution:
    def findDifferentBinaryString(self, nums: list[str]) -> str:
        nums.sort()
        n = len(nums)
        for num in range(1<<n):
            tar = bin(num)[2:]
            tar = "0" * (n - len(tar)) + tar
            left, right = 0, n-1
            find = False
            while left <= right:
                mid = (left + right) // 2
                if nums[mid] < tar:
                    left = mid + 1
                elif nums[mid] > tar:
                    right = mid - 1
                else:
                    find = True
                    break
            if not find:
                return tar
        return "0" * n

File: Cpp/1980_find_unique_binary_string.cpp
// https://leetcode.cn/problems/find-unique-binary-string/

#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
#include <bitset>

using namespace std;

class Solution {
public:
    string findDifferentBinaryString(vector<string>& nums) {
        sort(nums.begin(), nums.end());
        int n = nums.size();

        auto to_bin = [&n](int num) {
            vector<char> bins(n, '0');
            string res;
            int i = n-1;
            while (num > 0) {
                bins[i] = '0' + num % 2;
                i -= 1;
                num /= 2;
            }
            for (const auto &ch: bins) {
                res += ch;
            }
            return res;
        };

        for (int num = 0; num < (1 << n); ++num) {
            string tar = to_bin(num);
            int left = 0, right = n-1;
            bool find = false;
            while (left <= right) {
                int mid = (left + right) / 2;
                if (nums[mid] < tar) left = mid + 1;
                else if (nums[mid] > tar) right = mid - 1;
                else {
                    find = true;
                    break;
                }
            }
            if (!find) return tar;
        }
        return to_bin(0);
    }
};

int main() {
    Solution sol;
    vector<string> nums = {"01", "10"};
    cout << sol.findDifferentBinaryString(nums) << '\n';
    return 0;
}